src/main/java/com/google/sps/servlets/FirebaseAuthVerification.java

```diff
@@ -26,53 +26,35 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.apache.commons.io.IOUtils;
-
 import java.io.*;
 import java.util.List;
-
 import com.google.auth.oauth2.GoogleCredentials;
 import com.google.firebase.auth.FirebaseAuthException;
-import java.io.*;
+
+import com.google.sps.servlets.FirebaseAppInit;
 
 /* Servlet that takes an idToken, and then calls firebase to decode it and provide a uid if the token is valid. */
 @WebServlet("/idTokenVerification")
 public class FirebaseAuthVerification extends HttpServlet {
 
-    FirebaseToken decodedToken;
     FirebaseToken decodedUseridToken;   
     
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException  {
 
-        InputStream serviceAccount = FirebaseAuthVerification.class.getResourceAsStream("/firebaseServiceAccount.json");
-
-        FirebaseOptions options = new FirebaseOptions.Builder()
-            .setCredentials(GoogleCredentials.fromStream(serviceAccount))
-            .setDatabaseUrl("https://team147-step2020.firebaseio.com")
-            .build();
+        FirebaseAppInit firebaseAppInit = new FirebaseAppInit();
+        FirebaseApp firebaseApp = firebaseAppInit.initializeFirebaseApp();
```

Comment:
This is going to trigger initializeFirebaseApp everytime a new request coming. suggest storing the FirebaseApp as an attribute in FirebaseAuthVerification

Reply:
I noticed that we will always be initializing firebase every time we make a call to this class. However, i could not find a way (in java) to store the FirebaseApp once and for all, so I changed the function initializeFirebaseApp() to make a new instance only if a default instance has not been initialized. This way, FirebaseApp is initialized once, and never re-initializes even when the function is called multiple times.
The function is between line 50 - 61 in FirebaseAppInit.java. If there is a way to store the attribute once, i think that would be a better approach

Reply:
moved the initialization to constructor